lib/src/main/java/io/ably/lib/realtime/AblyRealtime.java

```diff
@@ -148,21 +143,24 @@ public Channel get(String channelName) {
         }
 
         @Override
-        public Channel get(String channelName, ChannelOptions channelOptions) throws AblyException {
-            Channel channel = map.get(channelName);
-            if (channel != null) {
+        public Channel get(final String channelName, final ChannelOptions channelOptions) throws AblyException {
+            // We're not using computeIfAbsent because that requires Java 1.8.
+            // Hence there's the slight inefficiency of creating newChannel when it may not be
+            // needed because there is an existingChannel.
+            final Channel newChannel = new Channel(AblyRealtime.this, channelName, channelOptions);
+            final Channel existingChannel = map.putIfAbsent(channelName, newChannel);
```

Comment:
You can use `map.get(channelName) `for concurrentHashmap too right?
https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ConcurrentHashMap.html#get(java.lang.Object)

Reply:
I don't understand your point. That is the method that this refactor is pretty much focussed on avoiding the use of. I've replaced two calls (one to `get`, followed by a subsequent conditional call to `put`, with a single call to atomic `putIfAbsent` as backed by the concurrent hash map).

Perhaps you need to re-read the issue to see what we're aiming to solve here. 🤔

Reply:
Actually, I was looking at the comment `slight inefficiency of creating newChannel`, if we use `map.get(channelName)`, we can avoid it. I feel it's a tradeoff, in most cases, the client will get an existing channel rather than creating a new one.